### PR TITLE
Refactor cli config to simplify language and document path parsing

### LINKED ISSUE #125: Documentation is lacking and I have a hard time figuring out how to use Accent
Hi!

I tried accent with a Vue frontend using vue-i18n with English as the master language and German as a translation.

The first thing I tried was very naive. I assumed you have a json file for every language and tried to make accent read and write the same file.

```
{
    "language": "en",
    "format": "json",
    "source": "src/locales/en.json",
    "target": "src/locales/en.json"
},
{
    "language": "de",
    "format": "json",
    "source": "src/locales/de.json",
    "target": "src/locales/de.json"
}
```

I have 6 translation ids right now and this made me translate 12 strings per language. Obviously wrong and since there's a source and a target parameter, not surprising. Then I tried having one file with empty strings for every id and use that as a source with different targets

```
{
    "language": "en",
    "format": "json",
    "source": "src/transkeys.json",
    "target": "src/locales/en.json"
},
{
    "language": "de",
    "format": "json",
    "source": "src/transkeys.json",
    "target": "src/locales/de.json"
}
```

Now I had 6 strings for every language. Great! But the documentation uses a glob. But how does it merge files? So I tried this

```
{
    "language": "en",
    "format": "json",
    "source": "src/trans/*.json",
    "target": "src/locales/en.json"
},
{
    "language": "de",
    "format": "json",
    "source": "src/trans/*.json",
    "target": "src/locales/de.json"
}
```

And I had 2 json files in trans with 2 keys each. That worked as well! 4 string per language. But then when I synced and exported, I only got 2 keys in de.json and en.json. My assumption: It doesn't merge the files and it's more for translation systems that have multiple files. Makes sense since I feel like this is more popular in large projects.

So back to my second attempt with one source file and multiple target files. I translated every string, exported and had the same German strings in both en.json and de.json.

I have no idea why and the documentation doesn't really tell me why either. Neither does the CLI nor does the documentation of the CLI. I have no idea how to use accent at all and everything that makes sense to me based on reading the documentation was a failure.

I'd really like to use Accent especially because I can't justify running something as hungry as weblate (which I know from work) and I love the CLI that comes with it. But I have no idea how to do that and I feel like the documentation shouldn't leave me hanging this early when trying out Accent.

### PR DESCRIPTION
**Pull request by simonprev**
This pull request fixes some confusing concepts

Fixes #125

## No more confusing "language" config.

The `files` stored in the config represent source file. The source files always comes from the master language

Here is the example for an Elixir Phoenix project, with the english as the master language.

```
{
  "files": [
    {
      "format": "gettext",
      "source": "priv/en/*.po",
      "target": "priv/%slug%/%original_file_name%.po"
    }
  ]
}
```

## Support for `namePattern` as document path in Accent

There is 2 patterns:
- The filename is the document path used in Accent
	- Like `config/locales/en/admin.yml`, the document path will be `admin`
- The filename is the language inside a folder
	- Like `app/translations/en.json`, the document path will be `translations`.

The pattern is inferred by the CLI but also can be specified in the config.

## Create directories on write

No need to create folders for target languages before using the `--write` flag.
In case of a file like `app/translations/fr/admin.json`, the CLI will automatically create `app/translations/fr` if it does not exist.

## Useful logs

- Instead of printing the Accent document path used to write locally, the logs display the real path in your filesystem
- On `--add-translations`, if no target languages, it does not log an empty title.
- The `Add translations` log title includes all the target language slugs.

# 🎉 Result
image

### CODE REVIEW

File: cli/src/services/document.ts [hunk 1]
```diff
@@ -115,28 +129,51 @@ export default class Document {
...
+    const url = `${this.apiUrl}/jipt-export?${this.encodeQuery(query)}`;
     const response = await fetch(url, {
       headers: this.authorizationHeader()
     });
 
     return this.writeResponseToFile(response, file);
   }
 
+  private encodeQuery(params: string[][]) {
+    return params.map(([name, value]) => `${name}=${value}`).join('&');
```
Comment (remiprev) [inline cli/src/services/document.ts]: We should use `encodeURIComponent` here.

```suggestion
    return params.map(([name, value]) => `${name}=${encodeURIComponent(value)}`).join('&');
```

**Comment by remiprev** [comment]
Should we have a `CHANGELOG.md` file to document these changes?

**Comment by ro-kue** [comment]
Very nice. Thanks for fixing this. Do you know when this will be released?

**Comment by machty** [comment]
Not sure where the docs are hosted but https://www.accent.reviews/guides/getting-started.html still references the `language` key.

**Comment by machty** [comment]
Also, what is `namePattern`? Is it a new key/option?

**Comment by simonprev** [comment]
Yeah I have 1 bug left before releasing the new version and deploying the guides.

The `namePattern` is to tell the cli how to parse the file name to name the file in Accent.

From new guide that will be updated after the release:

```
## Name pattern

{
  "apiUrl": "http://your.accent.instance",
  "apiKey": "2nziVSaa8yUJxLkwoZA",
  "files": [
    {
      "namePattern": "file",
      "format": "json",
      "source": "app/locales/fr/*.json",
      "target": "app/locales/%slug%/%document_path%.json"
    }
  ]
}

### file (default)

The default document name parsing is `file`.

Given the source `app/locales/fr/*.json` and a file `app/locales/fr/admin.json`, the document path in Accent will be `admin`.

### parentDirectory

Some frameworks and tools stores all translations inside the same directory.

Given the source `app/locales/**/*.json` and a file `app/locales/admin/fr.json`, the document path un Accent will be `admin`.

### fullDirectory

Given the source `app/locales/**/*.json` and a file `app/locales/admin/fr.json`, the document path un Accent will be `app/locales/admin`.
```

**Pull request merged**